File: Domain/Users/Models/UserModel.cs
using System;
using Domain.Base;
using Domain.Shared.Enum;

namespace Domain.Users.Models
{
    public class UserModel : BaseModel
    {
        public string name { get; set; }
        public string lastName { get; set; }
        public string email { get; set; }
        public string password { get; set; }
        public DateTime? dateOfBirth { get; set; }
        public GenderEnum? gender { get; set; }
        public PronounEnum? pronoum { get; set; }
        public string customGender { get; set; }
    }
}

File: Web/ClientApp/src/app/shared/models/user.model.ts
import { BaseModel } from "./base.model";
import { GenderEnum } from "../enum/gender.enum";
import { PronounEnum } from "../enum/pronun.enum";

export class UserModel extends BaseModel {
  name: string;
  lastName: string;
  email: string;
  dateOfBirth: string;
  gender?: GenderEnum;
  pronoun?: PronounEnum;
  customGender?: string;

  constructor() {
    super();
    this.name = "";
    this.lastName = "";
    this.email = "";
    this.dateOfBirth = "";
    this.gender = null;
    this.pronoun = null;
    this.customGender = null;
  }
}
